examples/RecursionExample.elm

```diff
@@ -0,0 +1,31 @@
+module RecursionExample exposing (RecursiveType, init, view)
+
+import Debug.Control exposing (Control, choice, lazy, list, map, string, value, values)
+import Html exposing (Html)
+import Html.Attributes as Html exposing (style)
+import String
+
+
+type RecursiveType
+    = RecursiveType (Maybe RecursiveType)
+
+
+init : Control RecursiveType
```

Comment:
🎨 this should probably be called `recursiveTypeControl` instead of `init`, I guess, since controls are meant to feel more like json decoders than like models.  But I'll fix that for all the examples later.